backend/src/documentation/padding-oracle.md

```diff
@@ -0,0 +1,214 @@
+
+Pay attention to the following case.
+
+C2 block
+
+> 00 | 00 | 00 | **2D**
+
+Decrypted value
+
+> A8 | 09 | F7 | **01** 
+
+The last byte of C2 with the value of **2D** decrypted to a valid padding because the last byte of the decrypted value is **01**. According to the PKCS#7 standard, this is valid padding, so, we found the value we want!
+
+> But how can we use this information to decrypt the C3 block?
+
+Now is the easy part. We know the C1 last byte outputs valid padding (**2D**) and we know the plaintext output which is **01**. Now we can infer the value of X, the output of D(E,C3). If you have noticed, we can do that because the X value XORed with C2 outputs the plaintext. At this point, you may know XOR is a commutative operation. 
+
+If,
+
+> Plaintext = C2 XOR X
+
+Then,
+
+> C2 XOR Plaintext = X
+
+And that's how we get the value of X (D(K,C3)).
+
+In our example,
+
+> 0x01 = 0x2D XOR X
+
+So,
+
+> X = 0x2D XOR 0x01
+
+Finally,
+
+> X = 0x2C
+
+0x01 is plaintext[15] and 0x2D is C2[15].
+
+We know the value of X (2C) and we are able to deduce the final value of the plaintext last byte. Simply XOR X with the previous original ciphertext block which is **6A211E23**. Confused? Compare these steps with the diagram of the CBC decryption shown above.
+
+At this time, the last byte of C3 is known; to find the rest of it, we can work backward through the entire block until every byte of X function is cracked, thus letting us decrypt the C3 plaintext one byte at a time. For the other blocks, it is just applying the same method to crack the full message.
+
+## Put the attack in practice
+
+There are a bunch of available scripts over the internet to automate this attack. We will using [this](https://github.com/KishanBagaria/padding-oracle-attacker) tool to get the flag. 
+
+According to the tool documentation, we need the **target URL**, **ciphertext encoding**, **block size** and the **HTTP code error**.
+
+Example:
+
+
+> poattack encrypt http://localhost:2020/decrypt?ciphertext= hex:666f6f2062617220f09fa684 16 400
+
+* **poattack** - name of the program
+* **encrypt** -  the operation
+* **http://localhost:2020/decrypt?ciphertext=** - URL
+* **hex:666f6f2062617220f09fa684** - encoding and payload
+* **16** - block size
+* **400** - HTTP error
+
+In the end execution, the output should the similar to:
+
+![poattack script output](/documentation/img/poattack.png "poattack script output")
+
+Now you have the flag!
+
+
+## Lesson Learned
+
+* A Padding Oracle Attack is a well known devastating attack against CBC and it will try to **break anything that uses CBC mode**.
+
+* In the context of an application, it depends on how the user implementation of CBC mode because, if an attacker can modify the ciphertext or can trigger an error, there are high chances he can break the encrypted message.
+
+* [Encrypt and MAC your data](https://security.stackexchange.com/questions/38942/how-to-protect-against-padding-oracle-attacks).
```

Comment:
lets try to use more 'trustworthy' sources like owasp...
Also, you can add other encryption schemas like GCM that handle the MAC behind scenes.
Finally may also be worth it to explain that before decrypting mac should be validated